File: Ui/UiXamarin/App.Droid/MainActivity.cs
using System;
using System.Collections.Generic;
using Android.App;
using Android.OS;
using Android.Support.Design.Widget;
using Android.Support.V4.Widget;
using Android.Support.V7.App;
using Android.Support.V7.Widget;
using Android.Views;
using Android.Views.Animations;
using Android.Widget;
using Toolbar = Android.Support.V7.Widget.Toolbar;

namespace App.Droid
{
    [Activity(Theme = "@style/AppTheme.NoActionBar", MainLauncher = true)]
    public class MainActivity : AppCompatActivity, View.IOnClickListener, SwipeRefreshLayout.IOnRefreshListener
    {
        private UsersAdapter _usersAdapter;
        private SwipeRefreshLayout _refreshLayout;

        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            SetContentView(Resource.Layout.activity_main);

            var toolbar = FindViewById<Toolbar>(Resource.Id.toolbar);
            SetSupportActionBar(toolbar);

            var fab = FindViewById<FloatingActionButton>(Resource.Id.fab);
            fab.SetOnClickListener(this);

            var rv = FindViewById<RecyclerView>(Resource.Id.rv_users);
            rv.SetLayoutManager(new LinearLayoutManager(this));
            rv.SetAdapter(_usersAdapter = new UsersAdapter { Users = GetUsers() });

            _refreshLayout = FindViewById<SwipeRefreshLayout>(Resource.Id.refresh_users);
            _refreshLayout.SetOnRefreshListener(this);
        }

        public override bool OnCreateOptionsMenu(IMenu menu)
        {
            MenuInflater.Inflate(Resource.Menu.menu_main, menu);
            return true;
        }

        public override bool OnOptionsItemSelected(IMenuItem item)
        {
            var id = item.ItemId;
            if (id == Resource.Id.action_settings)
            {
                return true;
            }

            return base.OnOptionsItemSelected(item);
        }

        public void OnClick(View v)
        {
            if (v.Id == Resource.Id.fab)
            {
                Snackbar.Make(v, "Replace with your own action", Snackbar.LengthLong).SetAction("Action", listener: null).Show();
            }
        }

        private static List<User> GetUsers()
        {
            var users = new List<User>();

            for (var i = 0; i < 200; i++)
            {
                var value = new Random().Next(0, 999999);
                users.Add(new User
                {
                    FirstName = $"FirstName{value}",
                    LastName = $"LastName{value}",
                    Email = $"Email@Email{value}"
                });
            }

            return users;
        }

        public void OnRefresh()
        {
            _usersAdapter.Users = GetUsers();
            _usersAdapter.NotifyDataSetChanged();
            _refreshLayout.Refreshing = false;
        }
    }

    internal class UsersAdapter : RecyclerView.Adapter
    {
        public List<User> Users { get; set; }

        public override void OnBindViewHolder(RecyclerView.ViewHolder holder, int position)
        {
            var vh = (UserViewHolder) holder;
            var user = Users[position];

            holder.ItemView.StartAnimation(AnimationUtils.LoadAnimation(holder.ItemView.Context, Resource.Animation.anim_fade_fall));

            vh.User = user;
        }

        public override RecyclerView.ViewHolder OnCreateViewHolder(ViewGroup parent, int viewType)
        {
            return new UserViewHolder(parent);
        }

        public override int ItemCount => Users?.Count ?? 0;
    }

    internal class UserViewHolder : RecyclerView.ViewHolder
    {
        private readonly ImageView _iconView;
        private readonly TextView _labelView;

        private User _user;

        public User User
        {
            get => _user;
            set
            {
                _user = value;
                LoadView();
            }
        }

        public UserViewHolder(ViewGroup parent) : base(LayoutInflater.From(parent.Context).Inflate(Resource.Layout.view_user, parent, false))
        {
            _iconView = ItemView.FindViewById<ImageView>(Resource.Id.iv_icon);
            _labelView = ItemView.FindViewById<TextView>(Resource.Id.tv_label);
        }

        private void LoadView()
        {
            _labelView.Text = $"{_user.FirstName} {_user.LastName}\n{_user.Email}";
        }
    }
}

File: Ui/UiAndroid/app/src/main/java/com/bpxamarin/uiandroid/MainActivity.java
package com.bpxamarin.uiandroid;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.view.LayoutInflater;
import android.view.View;
import android.view.Menu;
import android.view.MenuItem;
import android.view.ViewGroup;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class MainActivity extends AppCompatActivity implements SwipeRefreshLayout.OnRefreshListener {

    private UsersAdapter usersAdapter;
    private SwipeRefreshLayout refreshLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        FloatingActionButton fab = findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                        .setAction("Action", null).show();
            }
        });

        RecyclerView rv = findViewById(R.id.rv_users);
        rv.setLayoutManager(new LinearLayoutManager(this));
        rv.setAdapter(usersAdapter = new UsersAdapter());
        usersAdapter.setUsers(getUsers());

        refreshLayout = findViewById(R.id.refresh_users);
        refreshLayout.setOnRefreshListener(this);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();

        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }


    private static List<User> getUsers()
    {
        List<User> users = new ArrayList<>();

        for (int i = 0; i < 200; i++)
        {
            int value = new Random().nextInt(999999);
            User user = new User();
            user.setFirstName("FirstName"+value);
            user.setFirstName("LastName"+value);
            user.setFirstName("Email@Email"+value);
            users.add(user);
        }

        return users;
    }

    @Override
    public void onRefresh() {
        usersAdapter.setUsers(getUsers());
        usersAdapter.notifyDataSetChanged();
        refreshLayout.setRefreshing(false);
    }

    public class UsersAdapter extends RecyclerView.Adapter<UserViewHolder>{

        private List<User> users;

        public void setUsers(List<User> users){
            this.users = users;
        }

        @NonNull
        @Override
        public UserViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
            return new UserViewHolder(viewGroup);
        }

        @Override
        public void onBindViewHolder(@NonNull UserViewHolder viewHolder, int i) {
            viewHolder.itemView.startAnimation(AnimationUtils.loadAnimation(viewHolder.itemView.getContext(), R.anim.anim_fade_fall));
            User user = users.get(i);
            viewHolder.setUser(user);
        }

        @Override
        public int getItemCount() {
            return users != null ? users.size() : 0;
        }
    }

    private class UserViewHolder extends RecyclerView.ViewHolder{

        private ImageView iconView;
        private TextView labelView;

        private User user;

        UserViewHolder(@NonNull ViewGroup parent) {
            super(LayoutInflater.from(parent.getContext()).inflate(R.layout.view_user, parent, false));
            iconView = itemView.findViewById(R.id.iv_icon);
            labelView = itemView.findViewById(R.id.tv_label);
        }

        public void setUser(User user){
            this.user = user;
            loadView();
        }

        private void loadView(){
            labelView.setText(user.getFirstName() + " " + user.getLastName() + "\n" + user.getEmail());
        }
    }
}
